### PR TITLE
feat(contentful): add purchase common query

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- Need to query Purchase Details and Common Content for an array of stripe plan ids.

## This pull request

- Adds PurchaseWithDetailsOfferingContentUtil to retrieve the necessary data from Contentful for a specific locale.

## Issue that this pull request solves

Closes: #FXA-8537

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/types.ts
Comment (StaberindeZA) [inline libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/types.ts:42]: These names feel a little unwieldy. Should we rename it to something simpler, or does the long name provide useful context?

File: libs/shared/contentful/src/lib/contentful.manager.ts
Comment (xlisachan) [inline libs/shared/contentful/src/lib/contentful.manager.ts:23]: Nit: non-blockers 
- `PurchaseWithDetailsOfferingContentUtil` should be able to be imported from `./queries/purchase-with-details-offering-content` as well?
- order of imports (after L16)

File: libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/factories.ts
Comment (xlisachan) [inline libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/factories.ts:1]: Nit: non-blocker - should this file be a part of `libs/shared/contentful/src/lib/factories.ts` or should we break apart that file and have factories within their corresponding query? I prefer the latter.

File: libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/factories.ts
Comment (xlisachan) [inline libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/factories.ts:43]: Per Ecosystem Docs, the values for newsletterLabelTextCode are: `snp`, `hubs`, `mdnplus`. Should this be `faker.helpers.arrayElements(['snp', 'hubs', 'mdnplus']),`?
Ref: [https://mozilla.github.io/ecosystem-platform/tutorials/subscription-platform#stripe-product-metadata](https://mozilla.github.io/ecosystem-platform/tutorials/subscription-platform#stripe-product-metadata)

File: libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/factories.ts
Comment (xlisachan) [inline libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/factories.ts:44]: Per Ecosystem Docs, the values for newsletterSlug are: `mozilla-accounts`, `security-privacy-news`, `hubs`, `mdnplus`. 
Should this be `faker.helpers.arrayElements(['mozilla-accounts', 'security-privacy-news', 'hubs', 'mdnplus']),`?
Ref: [https://mozilla.github.io/ecosystem-platform/tutorials/subscription-platform#stripe-product-metadata](https://mozilla.github.io/ecosystem-platform/tutorials/subscription-platform#stripe-product-metadata)

File: libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/types.ts
Comment (xlisachan) [reply libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/types.ts:43]: I'll defer to the team.

**Review by xlisachan — APPROVED** [review]
Thanks Reino! I left a couple of comments and non-blocking nits. Otherwise, LGTM! (Also, I wouldn't mind if another reviewer took a look).

File: libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/factories.ts
Comment (xlisachan) [reply libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/factories.ts:1]: Aah, I see Julian made a comment on another PR with the same preference. Please disregard this nit. :)

File: libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/types.ts
Comment (StaberindeZA) [reply libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/types.ts:43]: Will leave it to a polish PR. :)

**Pull request merged**
